base/linalg/lapack.jl

```diff
@@ -3635,6 +3691,61 @@ for (trsen, elty) in
             end
             T, Q, w
         end
+        function tgsen!(select::Array{Int}, S::StridedMatrix{$elty}, T::StridedMatrix{$elty},
+                                            Q::StridedMatrix{$elty}, Z::StridedMatrix{$elty})
+# *       .. Scalar Arguments ..
+# *       LOGICAL            WANTQ, WANTZ
+# *       INTEGER            IJOB, INFO, LDA, LDB, LDQ, LDZ, LIWORK, LWORK,
+# *      $                   M, N
+# *       DOUBLE PRECISION   PL, PR
+# *       ..
+# *       .. Array Arguments ..
+# *       LOGICAL            SELECT( * )
+# *       INTEGER            IWORK( * )
+# *       DOUBLE PRECISION   DIF( * )
+# *       COMPLEX*16         A( LDA, * ), ALPHA( * ), B( LDB, * ),
+# *      $                   BETA( * ), Q( LDQ, * ), WORK( * ), Z( LDZ, * )
+# *       ..
+            chkstride1(S, T, Q, Z)
+            n, nt, nq, nz = chksquare(S, T, Q, Z)
+            n==nt==nq==nz || throw(DimensionMismatch("matrices are not of same size"))
+            ld = max(1, n)
```

Comment:
You'd need an `ld` variable for each input matrix. If I define
```julia
julia> A = randn(3,3);As = sub(A, 1:2, 1:2)
2x2 SubArray{Float64,2,Array{Float64,2},(UnitRange{Int64},UnitRange{Int64})}:
  0.148009  -0.241772
 -0.43966    1.44392 

julia> size(As, 1)
2

julia> stride(As, 2)
3
```
The stride information tells LAPACK how many steps in memory it takes to get to the next column. For a `Matrix{<:BlasFloat}` it is the same as `size(Matrix, 1)`, but not for a `SubMatrix`.